tests/test_dynamic_target_chunks.py

```diff
@@ -0,0 +1,63 @@
+import dask.array as dsa
+import pytest
+import xarray as xr
+
+from pangeo_forge_recipes.aggregation import dataset_to_schema
+from pangeo_forge_recipes.dynamic_target_chunks import dynamic_target_chunks_from_schema
+
+
+class TestDynamicTargetChunks:
```

Comment:
This is potentially a good use case for hypothesis - you could parameterize a unit test with a hypothesis strategy that generates arbitrary (regular) chunks, then assert that the property that the returned target chunk size is within the specified tolerance.

Reply:
Yeah that is a good idea. I will have to dig into hypothesis a bit to understand how to encode the logic.

Reply:
This PR to dask might give you what you need (I can also finish it if that would help)

https://github.com/dask/dask/pull/9374